File: app/src/main/java/mariavv/fitnesspal/domain/Dish.java
package mariavv.fitnesspal.domain;

import android.support.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

public class Dish {

    @NonNull
    public Date date;
    @NonNull
    public String meal;
    @SerializedName("food_id")
    public int foodId;
    public int weight;

    public Dish(@NonNull Date date, @NonNull String meal, int foodId, int weight) {
        this.date = date;
        this.meal = meal;
        this.foodId = foodId;
        this.weight = weight;
    }
}

File: app/src/main/java/mariavv/fitnesspal/domain/Dish.kt
package mariavv.fitnesspal.domain

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import mariavv.fitnesspal.data.db.ColumnName
import mariavv.fitnesspal.data.db.TableName

@Entity(tableName = TableName.DISHES)
class Dish(
        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = ColumnName.ID)
        var id: Int = 0,
        @ColumnInfo(name = ColumnName.MEAL)
        var meal: String,
        @ColumnInfo(name = ColumnName.DATE)
        var date: Long,
        @ColumnInfo(name = ColumnName.FOOD_ID)
        var foodId: Int,
        @ColumnInfo(name = ColumnName.WEIGHT)
        var weight: Int) {

    constructor() : this(meal = "", date = 0, foodId = 0, weight = 0)
}
